### PR TITLE
PLAT-526: method will stop if obj state is Missing

### PR DESCRIPTION
**Pull request by MikhailZhukov**
Test that method call fails without deduplication step if object is in the Missing state

### CODE REVIEW

File: ledger-core/virtual/integration/method_test.go [hunk 1]
```diff
@@ -1351,3 +1351,64 @@ func Test_MethodCall_HappyPath(t *testing.T) {
...
+		CallFlags:      payload.BuildCallFlags(contract.CallIntolerable, contract.CallDirty),
+		Caller:         server.GlobalCaller(),
+		Callee:         objectRef,
+		CallSiteMethod: "Method",
+		CallOutgoing:   outgoing,
+	}
+
+	server.SendPayload(ctx, &pl)
+
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitStopOf(&execute.SMExecute{}, 1))
```
Comment (bigbes) [inline ledger-core/virtual/integration/method_test.go:1403]: extract WaitStopOf before SendPayload. Possible place for races

File: ledger-core/virtual/integration/method_test.go [hunk 2]
```diff
@@ -1351,3 +1351,64 @@ func Test_MethodCall_HappyPath(t *testing.T) {
 
 	mc.Finish()
 }
+
+func TestVirtual_Method_ForObjectWithMissingState(t *testing.T) {
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/method_test.go:1355]: maybe move this one to semiintegration test?

File: ledger-core/virtual/integration/method_test.go [hunk 3]
```diff
@@ -1351,3 +1351,64 @@ func Test_MethodCall_HappyPath(t *testing.T) {
...
+	t.Log("C5106")
+	mc := minimock.NewController(t)
+
+	server, ctx := utils.NewUninitializedServer(nil, t)
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/method_test.go:1359]: If you don't need to mock runner you can do just utils.NewServer(nil,t)

File: ledger-core/virtual/integration/method_test.go [hunk 4]
```diff
@@ -1351,3 +1351,64 @@ func Test_MethodCall_HappyPath(t *testing.T) {
...
+
+	server.IncrementPulseAndWaitIdle(ctx)
+
+	state := &payload.VStateReport{
+		Status: payload.Missing,
+		Object: objectRef,
+	}
+
+	server.SendPayload(ctx, state)
+	server.WaitActiveThenIdleConveyor()
```
Comment (bigbes) [inline ledger-core/virtual/integration/method_test.go:1390]: use server.Journal.WaitStopOf on VStateReport handler

File: ledger-core/virtual/integration/method_test.go (hunk 1 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/method_test.go]: done

File: ledger-core/virtual/integration/method_test.go (hunk 3 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/method_test.go]: done

File: ledger-core/virtual/integration/method_test.go (hunk 4 above)
Comment (MikhailZhukov) [reply ledger-core/virtual/integration/method_test.go]: done

File: ledger-core/virtual/integration/method_test.go [hunk 5]
```diff
@@ -1351,3 +1351,65 @@ func Test_MethodCall_HappyPath(t *testing.T) {
...
+		CallSiteMethod: "Method",
+		CallOutgoing:   outgoing,
+	}
+
+	server.SendPayload(ctx, &pl)
+
+	testutils.WaitSignalsTimed(t, 10*time.Second, execDone)
+	testutils.WaitSignalsTimed(t, 10*time.Second, server.Journal.WaitAllAsyncCallsDone())
+
+	require.Equal(t, 1, typedChecker.VCallResult.Count())
```
Comment (sexyhamster) [inline ledger-core/virtual/integration/method_test.go:1407]: typedCheck has MinimockWait method, so you won't need to explicitly check all possible messages count == 0

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/method_test.go:1407]: done

Comment (sexyhamster) [reply ledger-core/virtual/integration/method_test.go:1407]: I'd rather leave typedCheck.MinimockWait(20*time.Second) here. To ensure unhandled messages are forbidden.

Comment (MikhailZhukov) [reply ledger-core/virtual/integration/method_test.go:1407]: done, but set to 10 sec

**Comment by ita-sammann** [comment]
Test comment, just ignore it.

**Pull request merged**